### DISCUSSION TITLE
Can i expose whole nextjs app ?

### DISCUSSION THREAD

**Discussion by IrfanAli17899** [module-federation/module-federation-examples#1979] [Q&A]
just like in an iframe, the app runs fully independently can I expose my whole next app using this package? a component example is working fine for me, but I need to expose the whole next app so it should run independently just like an iframe. 

when i try to do it, i get invalid hook call on _app.tsx <Component {...pageProps} />

please help me.  
@ScriptedAlchemy

**Comment by ScriptedAlchemy** [comment]
Next is entry point based. Meaning that _app and it’s pages are their own entry points. Webpack doesn’t know about each file in pages. 

I generally don’t use _app at all and just expose the pages individually _app causes unknown problems since importing a federated page has no reference that _app exists or is needed

**Reply by IrfanAli17899** [reply]
then I guess I can't make it work same as an iframe anyhow, where the app will run independently with its own nested routing and all?

**Reply by ScriptedAlchemy** [reply]
Next doesn't currently do nested routing. So you'd be limited to what the framework can even do to begin with. 

It would be technically possible to pull in another apps _app on the fly - but routing & page loading would require some custom integration. 

Next has limits based on its entrypoint based router system. 

You'd have to register extra routes into its router at runtime or use slug routes as catch alls. 

Bottom line, it's harder to do iframe like behavior since next isn't designed well for modularity within its core.

**Comment by dreamfalcon** [comment]
I have the same question.
So if using a nextjs app is not advised, is here any option to have rested routing and layout support?
Maybe migrate to a non nextjs app with react-router-dom?
I'm using the @ScriptedAlchemy nextjs-mf plugin.

**Comment by ScriptedAlchemy** [comment]
Next.js support is ending. look at pinned issue
